### ISSUE TITLE
Breadcrumbs on ModelAdmin inspect view is visually broken

### ISSUE THREAD

**Issue by loicteixeira** [wagtail/wagtail#3889]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Breadcrumbs is partially hidden by the sidebar.
<img width="580" alt="modeladmin" src="https://user-images.githubusercontent.com/6834938/31053048-aa514ae8-a6d7-11e7-821e-b486cf277fac.png">

On small screens, breadcrumbs are on white background.
<img width="360" alt="modeladmin2" src="https://user-images.githubusercontent.com/6834938/31053194-5687c5a0-a6db-11e7-8638-ce6d1acf12bf.png">

### Steps to Reproduce

1. Create a model
2. Create the corresponding ModelAdmin register it.
3. Set `inspect_view_enabled` to `True` on that ModelAdmin
4. Navigate to the ModelAdmin index page.
5. Click `Inspect` on one of the instances.

**Comment by lb-** [comment]
I have investigated this and need some direction on how to proceed. @loicteixeira 


## Detailed Bug Description & Cause

The modeladmin templates [choose_parent](https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/modeladmin/templates/modeladmin/choose_parent.html) and [inspect](https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/modeladmin/templates/modeladmin/inspect.html) implement the page header as follows:

```
        {% block header %}
            {% include "modeladmin/includes/breadcrumb.html" %}
            {% include "wagtailadmin/shared/header.html" with title=view.get_page_title subtitle=view.get_page_subtitle icon=view.header_icon %}
        {% endblock %}
```

This results in the breadcrumb appearing outside the `header` tag, because shared/header.html has no way to easily insert a breadcrumb. Causing the following issues;

* Issue 1 - on screen size < 50em: breaks the access to the menu button and also the css background colour.
* Issue 2 - on screen size > 50em: the far left 'world' icon that is meant to take the user back to home is not visible.

In wagtailadmin, the work around is to not use the shared/header.html but put a custom header in each time and also use a template tag `explorer_breadcrumb`. For example:
 * [/wagtailadmin/pages/add_subpage.html](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailadmin/templates/wagtailadmin/pages/add_subpage.html) - uses the shared/header.html
* [/wagtailadmin/pages/edit.html](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailadmin/templates/wagtailadmin/pages/edit.html) - builds the header manually, not using shared/header.html

# Potential Solution 1 - fix it within modeladmin only

Emulate how wagtailadmin works around the shared/header by simply not using it, just writing custom html each time (for choose_parent and inspect).

Possibly reduce the need to write this twice by adding a new include such as modeladmin/includes/header_with_breadcrumb.html

Downside is that if the main header html structure changes, we will have to change it in this include in the future.

# Potential Solution 2 - edit shared/header.html

Edit wagtailadmin/shared/header.html to accept a variable such as `breadcrumb_items` add the following just under `<header class="...`.

```
{% if breadcrumb_items %} <ul class="breadcrumb">{{ breadcrumb_items }}</ul>{% endif %}
<ul class="breadcrumb">
    {% for item in breadcrumb_items %}
    <li class="{{ item.li_classes }}"><a href="{{ item.url }}" class="{% if item.icon %}icon icon-{{ item.icon }}{% endif %}{% if item.text_replace %} text-replace{% endif %}">{{ item.title }}</a></li>
    {% endfor %}
</ul>
{% endif %}
```

`breadcrumb_items` will need to be a list of items each with the following attributes used above, I will need to compare this use. We would generate this with a template tag or insert it into the context for the modeladmin choose_parent and inspect templates.

Downsides is that this is a more complicated fix and the breadcrumb_items would only be used for modeladmin templates currently. However, it leaves an avenue for future generalisation of the wagtailadmin templates where the header html is being rewritten multiple times.

Longer list of places where shared/header.html is not used and the header is generated differently each time.
* /wagtailadmin/templates/pages/create.html
* /wagtailadmin/templates/pages/edit.html
* /wagtailadmin/templates/pages/create.html
* /wagtailadmin/templates/pages/index.html

**Comment by loicteixeira** [comment]
Thanks for the detailed investigation @lb-.

While I can appreciate solution 2 for trying to make it right, I'm not sure this is the way to go since only modeladmin will use it. It basically bets that there will be future generalisation (which might not happen) and since the admin templates won't use it, someone looking for an example probably won't know it exists and don't use it either. Unless we're ready to tackle the generalisation now (or right after), leaving a half-backed solution might cause more trouble than solve problems.

On the other hand solution 1 doesn't improve anything but does fix the. Obviously not the greatest but for the reasons explained above, it has my preference.

**Comment by lb-** [comment]
Sounds good, that makes sense. I will see if I can get on that this week.

**Comment by gasman** [comment]
Fixed in #4029
